lib/compiler_options.cmake

```diff
@@ -15,13 +15,16 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 function(set_compile_options target)
+	# not sure why but I can only get intellisense in VS2019 working with -std=c++17
```

Comment:
The if(MSVC) test doesn't help as we're using Cmake with the ESP-IDF tool set so the compiler is GNU 8.2.0 and not MSVC.

Removing this as it appears to be fixed in VS2019 16.3 Preview 3.

Reply:
Hm, did you write that this was fixed in an update of VS?

Anyway, since ESP-IDF doesn't use MSYs any more, I think the way to wrap this using [CMAKE_HOST_SYSTEM_NAME](https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_NAME.html#variable:CMAKE_HOST_SYSTEM_NAME)